pandas/sparse/frame.py

```diff
@@ -302,7 +302,7 @@ def fillna(self, value=None, method=None, axis=0, inplace=False,
     # ----------------------------------------------------------------------
     # Support different internal representation of SparseDataFrame
 
-    def _sanitize_column(self, key, value):
+    def _sanitize_column(self, key, value, broadcast=False):
```

Comment:
I checked the following, which worked as expected:
```
df1 = pd.DataFrame({'a': [1, 2, 3]})
df2 = pd.DataFrame({'b': [2,3,4]})
df = pd.concat([df1, df1, df2], axis=1).to_sparse()
df['a'] = [1,2,5]
```
_sanitize_columns doesn't do broadcasting, but the SparseDataFrame setter seems to handle this case properly.

When playing around with SparseDataFrames, I did hit the same error in _box_item_values, but I was going to open a separate issue for that when I got home. Couldn't really figure out why that if/else clause was there to begin with, even after looking back at the commit history.